### PR TITLE
Core/Spark: Fix kryo deserialization of `SerializableTable`

### PR DESCRIPTION
**Pull request by Kontinuation**
Closes https://github.com/apache/iceberg/issues/5974

### CODE REVIEW

**Review by nastra — APPROVED** [review]

**Review by singhpk234 — APPROVED** [review]

File: core/src/test/java/org/apache/iceberg/hadoop/TestTableSerialization.java [hunk 1]
```diff
@@ -51,6 +52,15 @@ public void testSerializableTable() throws IOException, ClassNotFoundException {
     table.updateSchema().addColumn("new_col", Types.IntegerType.get()).commit();
 
     TestHelpers.assertSerializedAndLoadedMetadata(table, TestHelpers.roundTripSerialize(table));
+    Table serializableTable = SerializableTable.copyOf(table);
+    TestHelpers.assertSerializedAndLoadedMetadata(
+        serializableTable, TestHelpers.KryoHelpers.roundTripSerialize(serializableTable));
+
+    table.newAppend().appendFile(FILE_A).commit();
```
Comment (szehon-ho) [inline core/src/test/java/org/apache/iceberg/hadoop/TestTableSerialization.java:59]: Nit: would make a new test as it seems unrelated to previous one (ie, testSerializableTableWithSnapshot)

**Review by szehon-ho — APPROVED** [review]
Minor suggestion, but looks good to me, thanks!

File: core/src/test/java/org/apache/iceberg/hadoop/TestTableSerialization.java (hunk 1 above)
Comment (Kontinuation) [reply core/src/test/java/org/apache/iceberg/hadoop/TestTableSerialization.java:62]: Moved serialization of table with snapshot to a new test.

**Review by aokolnychyi — APPROVED** [review]

**Comment by aokolnychyi** [comment]
Thanks for fixing it, @Kontinuation! Thanks for reviewing, @nastra @singhpk234 @szehon-ho!
